src/robot_dart/robot.hpp

```diff
@@ -169,11 +177,21 @@ namespace robot_dart {
         void _set_color_mode(dart::dynamics::MeshShape::ColorMode color_mode, dart::dynamics::SkeletonPtr skel);
         void _set_color_mode(dart::dynamics::MeshShape::ColorMode color_mode, dart::dynamics::ShapeNode* sn);
 
+        Eigen::VectorXd _get_dof_data(int content, const std::vector<std::string>& dof_names = {}, bool filter_mimics = false, size_t start_dof = 0);
+        void _set_dof_data(int content, const Eigen::VectorXd& data, const std::vector<std::string>& dof_names = {}, bool filter_mimics = false, size_t start_dof = 0);
+        Eigen::VectorXd _get_vector_mimic(size_t start_dof, const Eigen::VectorXd& vec) const;
+        Eigen::VectorXd _set_vector_mimic(size_t start_dof, const Eigen::VectorXd& vec) const;
+
         std::string _robot_name;
         dart::dynamics::SkeletonPtr _skeleton;
         std::vector<RobotDamage> _damages;
         std::vector<std::shared_ptr<control::RobotControl>> _controllers;
-
+        std::unordered_map<std::string, size_t> _controllable_dof_map;
```

Comment:
Why do we need 3 different maps? One for controllable, one for full, and one for mimic joints? I would only have one map. What is the additional value that they are adding? What would be a use-case for this?

Reply:
Because I was creating the _controllable_dof_map it  was easy to create the _mimic_dof_map
And maybe the user will be interested to query those maps in an application

Reply:
My question was why do we need the `_controllable_map`, and why the `_full_dof_map` is not enough? Since this is a mapping from `names` to `indices`, why would we need different maps for the controllable joints and for the rest of the joints?

Maybe we need a function to get the names of the controllable/mimic/all dofs and not different maps?

Reply:
Yes this is a way for the user to get the names and index of the controllable/mimic/all dofs but we can also have a function to get the names only (instead of the different maps)